src/repositories/pog-profile.repository.ts

```diff
@@ -26,23 +26,32 @@ export class PogProfileRepository extends DefaultCrudRepository<
   }
 
   async createPogProfile(
-    pogProfile: Partial<PogProfile>,
+    pogProfileParams: Partial<PogProfile>,
   ): Promise<Partial<PogProfile> | undefined> {
+    const pogProfile: Partial<PogProfile> = {...pogProfileParams};
     const profileId = utils.formatBytes32String(nanoid());
     console.log('Creating new profile with userId:', profileId);
     Object.assign(pogProfile, {userId: profileId});
     const pogProfileRef = this.db.doc(`pogprofiles/${profileId}`);
 
+    // if email was provided but not email hash, generate email hash
+    if (!!pogProfile.email && !pogProfile.emailHash) {
+      Object.assign(pogProfile, {emailHash: this.hashEmail(pogProfile.email)});
+    }
+
+    // remove doGooder from pogprofile doc since it will go into the walletAddresses subcollection
+    delete pogProfile.doGooder;
```

Comment:
THIS is a smell that we're using the type for more than it is supposed to. I let it for now, but we might need to go for a better convertion or type definition